app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1606,6 +1616,13 @@ class MessagingManager @Inject constructor(
 
         if (uri.startsWith(SETTINGS_PREFIX) && uri.substringAfter(SETTINGS_PREFIX) == NOTIFICATION_HISTORY)
             intent.putExtra("fragment", NOTIFICATION_HISTORY)
+        if (uri.startsWith(INTENT_PREFIX) || uri.startsWith(APP_PREFIX) || uri.startsWith(DEEP_LINK_PREFIX))
+            intent.`package`?.let {
+                context.packageManager.getLaunchIntentForPackage(
+                    it
+                )
+            }
```

Comment:
Is this required? You don't seem to be using the output and intent, app and deep links seem to work fine without them.

Reply:
This was carried over from WebViewActivity, I can't remember why I had used it previously but I dont think it hurts to set the package explicitly if we can?

Reply:
Looking at the docs for [setPackage()](https://developer.android.com/reference/android/content/Intent#setPackage(java.lang.String)) I see now that it was added in cases where it might be needed.

```
(Usually optional) Set an explicit application package name that limits the components this Intent will resolve to.
```

"(Usually optional)" 😂

Reply:
I don't see where you're setting it on the intent though - or are the backticks similar to `=` somehow?

Reply:
Oh I see now, yes you are right its not used here 🤦‍♂️ will remove

Reply:
Actually now I remember, this was used in `WebViewActivity` to detect when an app did not exist on the device to load the play store so the user can install the app and consume the link. Will add it back properly. Thanks for spotting this one too. 😓